src/format.rs

```diff
@@ -44,3 +46,129 @@ pub fn extract_root_table(
     .map_err(|err| ConfigError::invalid_root(uri, err))
     .map_err(|err| Box::new(err) as Box<dyn Error + Send + Sync>)
 }
+
+// Equivalent to ValueKind, except Table + Array store the same enum
+// Useful for serde to serialize values into, then convert to Value.
+// NOTE: Order of variants is important. Serde will use whichever
+// the input successfully deserializes into first.
```

Comment:
As this just replaces our existing type we are deserializing to, I'll give this a pass.  Overall, I'd rather we deserialize directly into our needed type.

That would either require
- Putting a dummy value in for the origin
- Building up my `serde-basepath` idea I've been toying with, see https://github.com/clap-rs/clap/discussions/2763#discussioncomment-13915276